### ISSUE TITLE
False positive no-undef with destructuring assignment

### ISSUE THREAD

**Issue by izanf** [eslint/eslint#10958]
**Tell us about your environment**

* **ESLint Version:** ^5.6.1
* **Node Version:** v10.11.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
	"parser": "babel-eslint",
	"parserOptions": {
		"ecmaVersion": "6",
		"sourceType": "module",
		"ecmaFeatures": {
			"jsx": true,
			"modules": true
		}
	},
	"extends": [
		"standard",
		"plugin:react/recommended"
	],
	"settings": {
    "react": {
      "createClass": "createReactClass",
      "pragma": "React",
      "version": "16.3.1",
		},
	},
	"rules": {
		"semi": [2, "always"],
	},
	"globals": {
		"fetch": false
	}
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const userData = ({ login, email, phone, password, apps, profiles } = this.state);
```

```bash
    "lint": "./node_modules/.bin/eslint .",
```

**What did you expect to happen?**

He hoped that the destructuring of the object would be done smoothly as shown in the mozilla documentation. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

**What actually happened? Please include the actual, raw output from ESLint.**
'login' is not defined. (no-undef)
'email' is not defined. (no-undef)
'phone' is not defined. (no-undef)
'password' is not defined. (no-undef)
'apps' is not defined. (no-undef)
'profiles' is not defined. (no-undef)

**Comment by izanf** [comment]
@platinumazure

You made me think better, because I imagined that since it works eslint should accept, but I remembered that eslint helps to set standards and does not perform tests.

Thank you.
